tests/operator/operator.go

```diff
@@ -2644,48 +2644,45 @@ func waitForUpdateCondition(kv *v1.KubeVirt) {
 }
 
 func waitForKvWithTimeout(newKv *v1.KubeVirt, timeoutSeconds int) {
-	Eventually(func() error {
+	Eventually(func(g Gomega) {
 		kv, err := kubevirt.Client().KubeVirt(newKv.Namespace).Get(context.Background(), newKv.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
+		g.Expect(err).ToNot(HaveOccurred())
 
-		if kv.Status.Phase != v1.KubeVirtPhaseDeployed {
-			return fmt.Errorf("waiting for phase to be deployed (current phase: %+v)", kv.Status.Phase)
-		}
+		g.Expect(kv.Status.Phase).To(Equal(v1.KubeVirtPhaseDeployed),
+			"waiting for phase to be deployed (current phase: %+v)", kv.Status.Phase)
 
 		available := false
 		progressing := true
 		degraded := true
 		created := false
+
 		for _, condition := range kv.Status.Conditions {
-			if condition.Type == v1.KubeVirtConditionAvailable && condition.Status == k8sv1.ConditionTrue {
-				available = true
-			} else if condition.Type == v1.KubeVirtConditionProgressing && condition.Status == k8sv1.ConditionFalse {
-				progressing = false
-			} else if condition.Type == v1.KubeVirtConditionDegraded && condition.Status == k8sv1.ConditionFalse {
-				degraded = false
-			} else if condition.Type == v1.KubeVirtConditionCreated && condition.Status == k8sv1.ConditionTrue {
-				created = true
+			switch condition.Type {
+			case v1.KubeVirtConditionAvailable:
+				available = condition.Status == k8sv1.ConditionTrue
+
+			case v1.KubeVirtConditionProgressing:
+				progressing = condition.Status != k8sv1.ConditionFalse
+
+			case v1.KubeVirtConditionDegraded:
+				degraded = condition.Status != k8sv1.ConditionFalse
+
+			case v1.KubeVirtConditionCreated:
+				created = condition.Status == k8sv1.ConditionTrue
 			}
 		}
 
-		if !available || progressing || degraded || !created {
+		g.Expect(available && !progressing && !degraded && created).To(BeTrue(), func() string {
```

Comment:
This is kinda complex and if failure occurs I don't have overview if the upgrade was progressing or degraded and so on...